src/mongocxx/exception/private/error_category.cpp

```diff
@@ -382,7 +383,12 @@ class mongocxx_error_category_impl final : public std::error_category {
     }
 
     std::string message(int code) const noexcept override {
-        return "unknown mongocxx error";
+        switch(static_cast<error_code>(code)) {
+        case error_code::k_bulk_write_type_uninitialized:
+            return "bulk write type was k_uninitialized";
```

Comment:
I realize we're going to leave this class alone for now and we may not need this error, but if we do end up using it, I'd prefer the message be less opaque, something like "cannot operate on uninitialized bulk write"